cdap-ui/app/cdap/components/DataPrepHome/index.js

```diff
@@ -112,35 +114,68 @@ export default class DataPrepHome extends Component {
     this.eventEmitter.emit('DATAPREP_CLOSE_SIDEPANEL');
   }
 
-  updateWorkspaceList() {
-    let namespace = NamespaceStore.getState().selectedNamespace;
-
+  updateWorkspaceListRetry(namespace) {
     MyDataPrepApi.getWorkspaceList({ namespace })
-      .subscribe((res) => {
-        if (res.values.length === 0) {
+      .subscribe(
+        (res) => {
+          if (res.values.length === 0) {
+            this.setState({
+              isEmpty: true,
+              toggleConnectionsViewFlag: true,
+              backendDown: false,
+              backendCheck: false
+            });
+            DataPrepStore.dispatch({
+              type: DataPrepActions.disableLoading
+            });
+            return;
+          }
+          let sortedWorkspace = orderBy(res.values, [(workspace) => workspace.name.toLowerCase()], ['asc']);
+          let isCurrentWorkspaceIdValid = sortedWorkspace.find(ws => ws.id === this.props.match.params.workspaceId);
+          if (this.props.match.params.workspaceId && !isCurrentWorkspaceIdValid) {
+            let url = this.props.match.url.slice(0, this.props.match.url.indexOf(this.props.match.params.workspaceId));
+            this.props.history.replace(url);
+          }
           this.setState({
-            isEmpty: true,
-            toggleConnectionsViewFlag: true,
+            rerouteTo: sortedWorkspace[0].id,
             backendDown: false,
-            backendCheck: false
+            backendCheck: false,
+            currentWorkspaceId: sortedWorkspace[0].id
           });
-          return;
-        }
-        let sortedWorkspace = orderBy(res.values, [(workspace) => workspace.name.toLowerCase()], ['asc']);
-        let isCurrentWorkspaceIdValid = sortedWorkspace.find(ws => ws.id === this.props.match.params.workspaceId);
-        if (this.props.match.params.workspaceId && !isCurrentWorkspaceIdValid) {
-          let url = this.props.match.url.slice(0, this.props.match.url.indexOf(this.props.match.params.workspaceId));
-          this.props.history.replace(url);
+          DataPrepStore.dispatch({
+            type: DataPrepActions.disableLoading
+          });
+
+          this.fetching = false;
+        },
+        (err) => {
+          if (this.workspaceListRetries < 3) {
+            this.workspaceListRetries += 1;
+            this.updateWorkspaceListRetry(namespace);
+          } else {
+            this.setState({
+              backendDown: false,
+              backendCheck: false
+            });
+            DataPrepStore.dispatch({
+              type: DataPrepActions.disableLoading
+            });
+            DataPrepStore.dispatch({
+              type: DataPrepActions.setDataError,
+              payload: {
+                errorMessage: objectQuery(err, 'message') || objectQuery(err, 'response', 'message') || objectQuery(err, 'response')
```

Comment:
Same as the other comment. If this is not going to be used we might as well set `errorMessage` as `true`